packages/auth-construct/src/construct.ts

```diff
@@ -141,24 +146,41 @@ export class AmplifyAuth
     props: AuthProps = DEFAULTS.IF_NO_PROPS_PROVIDED
   ) {
     super(scope, id);
-
     this.name = props.name ?? '';
     this.domainPrefix = props.loginWith.externalProviders?.domainPrefix;
-
     // UserPool
     this.computedUserPoolProps = this.getUserPoolProps(props);
+
+    const userPoolProps = {
+      ...this.computedUserPoolProps,
+    };
+    let customSenderKmsKey: Key | undefined;
+    if (
+      props.senders?.email &&
+      props.senders.email instanceof lambda.Function
+    ) {
+      if (!customSenderKmsKey) {
+        customSenderKmsKey = new Key(this, 'CustomSenderKey', {
+          enableKeyRotation: true,
+        });
+      }
+      customSenderKmsKey.grantDecrypt(props.senders.email);
+      props.senders.email.addPermission('CognitoInvokeEmail', {
+        principal: new ServicePrincipal('cognito-idp.amazonaws.com'),
+        action: 'lambda:InvokeFunction',
```

Comment:
we should narrow down this policy to have conditions on account and user pool arn.

See https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-working-with-lambda-triggers.html how to do this.

if we don't this lambda will be callable from any user pool.